espoon-voltti/espooevents-service
# [ES-411]: Fix importers from overwriting audiences and keywords

#### Summary


**Fixed**

- The `helmet` importer (or any other importer for that matter) from removing Espoo audience keywords added by the `add_espoo_audience` management command and Espoo place keywords added by the `add_espoo_places` management command when reimporting and thus updating an existing event. This change fixes the issue by checking whether the event has any Espoo audiences or keywords and keeping them although the reimported event data doesn't have any Espoo audiences or keywords. This way, we don't accidentally overwrite any audiences and keywords for events that have been augmented by the `add_espoo_audience` and `add_espoo_places` management commands.

[ES-411]

#### Dependencies


None.

#### Testing instructions


1. Run the `helmet` importer
1. Run the `add_espoo_audience` and `add_espoo_places` management commands
1. Check for any events that were augmented with Espoo audiences and keywords and take a note of their event IDs
1. Run the `helmet` importer again
1. Verify that the events whose event IDs you took a note of still have the same Espoo audiences and keywords as before

#### Checklist for pull request creator


- [x] A task has been created for the PR on the Kanban board with necessary details filled (one task / repo)
- [x] The commits and commit messages adhere to [version control conventions](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit)
- [ ] ~The API's adhere to Voltti's REST API conventions~
- [ ] Tests have been written for the change (unit, REST API)
- [x] All code has been linted and there aren't any lint errors
- [x] The change has been tested locally, e.g., with httpie
- [ ] ~The change has been tested against a DB dump from test and/or staging if the models have been changed~
- [x] The changelog has been updated with the relevant changes and the version number has been incremented

#### Checklist for pull request reviewer (copy to review text box)


```
- [ ] A task has been created for the PR on the Kanban board with necessary details filled (one task / repo)
- [ ] The commits and commit messages adhere to [version control conventions](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit)
- [ ] The API's adhere to Voltti's REST API conventions
- [ ] Tests have been written for the change (unit, REST API)
- [ ] All code has been linted and there aren't any lint errors
- [ ] The change has been tested locally, e.g., with httpie
- [ ] The change has been tested against a DB dump from test and/or staging if the models have been changed
- [ ] The changelog has been updated with the relevant changes and the version number has been incremented
```


[ES-411]: https://voltti.atlassian.net/browse/ES-411
